src/librustc_lint/builtin.rs

```diff
@@ -272,6 +272,32 @@ impl EarlyLintPass for UnsafeCode {
                 })
             }
 
+            ast::ItemKind::Fn(..) => {
+                if attr::contains_name(&it.attrs, sym::no_mangle) {
+                    self.report_unsafe(cx, it.span, |lint| {
+                        lint.build("declaration of a `no_mangle` function").emit();
```

Comment:
Could we add a `note` to explain why the annotation is problematic and or a link to the docs?